lib/litmus_paper/dependency/script.rb

```diff
@@ -9,23 +11,27 @@ def initialize(command, options = {})
       end
 
       def available?
-        Timeout.timeout(@timeout) do
-          script_stdout = script_stderr = nil
-          script_status = POpen4.popen4(@command) do |stdout, stderr, stdin, pid|
-            @script_pid = pid
-            script_stdout = stdout.read.strip
-            script_stderr = stderr.read.strip
-          end
-          unless script_status.success?
-            LitmusPaper.logger.info("Available check to #{@command} failed with status #{$CHILD_STATUS.exitstatus}")
-            LitmusPaper.logger.info("Failed stdout: #{script_stdout}")
-            LitmusPaper.logger.info("Failed stderr: #{script_stderr}")
+        script_status = script_stdout = script_stderr = nil
+        Open3.popen3(@command, :pgroup=>true) do |stdin, stdout, stderr, wait_thr|
+          @script_pid = wait_thr.pid
+          thstderr = Thread.new { stderr.read }
+          thstdout = Thread.new { stdout.read }
+          if !wait_thr.join(@timeout) # wait thread does not end within timeout
```

Comment:
> why are you spawning a thread to read from std*, !wait_thr.join(@timeout) should ensure the other end of the std* pipes are closed so std* doesn't block indefinitely, if it blocks indefinitely thstd*.value will also block, so the thread is not really helping

@bjhaid moving to this pull request

This is so large writes to stdout and stderr do not block, I added a test to verify `it "is true when the script returns a lot of data"`